src/index.js

```diff
@@ -122,6 +122,14 @@ const definePropertiesFor = (WebComponent, mapping, onAfterSet) => {
 
 export function register(ReactComponent, tagName, mapping = {}) {
   const attributeNames = Object.keys(mapping);
+  const render = function() {
```

Comment:
yepp, `this` sucks, and we have to use it all over the place :/ Anyway, it's better to strive for purity, even if you can't make it a 100%. Refactored!